Assertion failure crash: assert specifier.conv_type is not None

**Crash Report**

The snippet below crashes mypy 1.19.1 and master (0cc21d99b).

**Traceback**

```python-traceback
Traceback (most recent call last):
  File ".venv-mypy/bin/mypy", line 7, in <module>
    sys.exit(console_entry())
  File "mypy/mypy/__main__.py", line 15, in console_entry
    main()
  File "mypy/mypy/main.py", line 135, in main
    res, messages, blockers = run_build(sources, options, fscache, t0, stdout, stderr)
  File "mypy/mypy/main.py", line 219, in run_build
    res = build.build(sources, options, None, flush_errors, fscache, stdout, stderr)
  File "mypy/mypy/build.py", line 320, in build
    result = build_inner(
  File "mypy/mypy/build.py", line 417, in build_inner
    graph = dispatch(sources, manager, stdout)
  File "mypy/mypy/build.py", line 3178, in dispatch
    process_graph(graph, manager)
  File "mypy/mypy/build.py", line 3588, in process_graph
    done, still_working, results = manager.wait_for_done(graph)
  File "mypy/mypy/build.py", line 1070, in wait_for_done
    process_stale_scc(graph, next_scc, self)
  File "mypy/mypy/build.py", line 3752, in process_stale_scc
    graph[id].type_check_first_pass()
  File "mypy/mypy/build.py", line 2543, in type_check_first_pass
    self.type_checker().check_first_pass()
  File "mypy/mypy/checker.py", line 530, in check_first_pass
    self.accept(d)
  File "mypy/mypy/checker.py", line 641, in accept
    stmt.accept(self)
  File "mypy/mypy/nodes.py", line 1602, in accept
    return visitor.visit_expression_stmt(self)
  File "mypy/mypy/checker.py", line 4898, in visit_expression_stmt
    expr_type = self.expr_checker.accept(s.expr, allow_none_return=True, always_allow_any=True)
  File "mypy/mypy/checkexpr.py", line 6081, in accept
    typ = self.accept_maybe_cache(node, type_context=type_context)
  File "mypy/mypy/checkexpr.py", line 6116, in accept_maybe_cache
    typ = node.accept(self)
  File "mypy/mypy/nodes.py", line 2452, in accept
    return visitor.visit_op_expr(self)
           ~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "mypy/mypy/checkexpr.py", line 3487, in visit_op_expr
    return self.strfrm_checker.check_str_interpolation(e.left, e.right)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "mypy/mypy/checkstrformat.py", line 681, in check_str_interpolation
    self.check_mapping_str_interpolation(specifiers, replacements, expr)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "mypy/mypy/checkstrformat.py", line 814, in check_mapping_str_interpolation
    assert specifier.conv_type is not None
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 
```

**To Reproduce**

```python
'%(a)' % {'a': 1}
```

**Your Environment**

- Mypy version used: 0cc21d99b (and 1.19.1)
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.13.11
- Operating system and version: Arch Linux

--- a/mypy/checkstrformat.py
+++ b/mypy/checkstrformat.py
@@ -145,10 +145,10 @@ class ConversionSpecifier:
         self.key = m_dict.get("key")
 
         # Replace unmatched optional groups with empty matches (for convenience).
-        self.conv_type = m_dict.get("type", "")
-        self.flags = m_dict.get("flags", "")
-        self.width = m_dict.get("width", "")
-        self.precision = m_dict.get("precision", "")
+        self.conv_type = m_dict.get("type") or ""
+        self.flags = m_dict.get("flags") or ""
+        self.width = m_dict.get("width") or ""
+        self.precision = m_dict.get("precision") or ""
 
         # Used only for str.format() calls (it may be custom for types with __format__()).
         self.format_spec = m_dict.get("format_spec")

--- a/test-data/unit/check-formatting.test
+++ b/test-data/unit/check-formatting.test
@@ -60,6 +60,7 @@ a = None # type: Any
 [case testStringInterpolationInvalidPlaceholder]
 '%W' % 1  # E: Unsupported format character "W"
 '%b' % 1  # E: Format character "b" is only supported on bytes patterns
+'%(a)' % {'a': 1}  # E: Unsupported format character ""
 
 [case testStringInterpolationWidth]
 '%2f' % 3.14

